samza-test/src/test/java/org/apache/samza/test/processor/TestStreamProcessor.java

```diff
@@ -132,7 +132,7 @@ public void testStreamProcessorWithNoTask() {
     configMap.remove("task.class");
     final Config configs = new MapConfig(configMap);
 
-    StreamProcessor processor = new StreamProcessor(1, configs, new HashMap<>());
```

Comment:
Not sure if there's extensive surgery needed in existing code.  But, it seems that there's probably work involved in writing the test. So, this is not a blocker.
- I think there is a constructor in `StreamProcessor` already takes in a taskFactory. 
- You can make the passed-in taskFactory to return an instance of `StreamTask` that throws an exception in its methods.